docs/education/dlt-fundamentals-course/lesson_6_how_dlt_works.py

```diff
@@ -0,0 +1,664 @@
+    # **Understanding `pipeline.run()`**
+     The `pipeline.run()` method executes the entire pipeline, encompassing the [`extract`](#scrollTo=4C0U1dnwZxAB), [`normalize`](#scrollTo=bCeUqaW_cRSh), and [`load`](#scrollTo=Rn6cUc0OcWsk) stages.
+    """)
+    return
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(
+        r"""![Lesson_6_How_dlt_works_img1](https://storage.googleapis.com/dlt-blog-images/dlt-fundamentals-course/Lesson_6_How_dlt_works_img1.png)"""
+    )
+    return
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(r"""Consider this intentionally short example:""")
+    return
+
+
+@app.cell
+def _():
+    import dlt
+
+    pipeline = dlt.pipeline(
+        pipeline_name="my_pipeline", destination="duckdb", progress="log"
+    )
+
+    load_info = pipeline.run(
+        [
+            {"id": 1},
+            {"id": 2},
+            {"id": 3, "nested": [{"id": 1}, {"id": 2}]},
+        ],
+        table_name="items",
+    )
+    print(load_info)
+    return
+
+
+@app.cell(hide_code=True)
+def _(mo):
+    mo.md(r"""
+    This is what happens when the `run` method is executed:
+
+    1. **Extract** - Fully extracts the data from your source to your hard drive. In the example above, an implicit source with one resource with 3 items is created and extracted.
+    2. **Normalize** - Inspects and normalizes your data and computes a schema compatible with your destination. For the example above, the normalizer will detect one column `id` of type `int` in one table named `items`, it will furthermore detect a nested list in table items and unnest it into a child table named `items__nested`.
+    3. **Load** - Runs schema migrations if necessary on your destination and loads your data into the destination. For the example above, a new dataset on a local duckdb database is created that contains the two tables discovered in the previous steps.
+
+
+    ## **Display the loading progress**
+    Notice how we use `progress="log"` here.
+
+    The `progress="log"` argument in the `dlt.pipeline` configuration enables detailed logging of the pipeline’s progress during execution. These logs provide visibility into the pipeline’s operations, showing how data flows through the **Extract**, **Normalize**, and **Load** phases. The logs include real-time metrics such as resource or file counts, time elapsed, processing rates, memory usage, and CPU utilization.
+
+
+    `dlt` supports 4 progress monitors out of the box:
+
+    * `enlighten` - a status bar with progress bars that also allows for logging.
+    * `tqdm` - the most popular Python progress bar lib, proven to work in Notebooks.
+    * `alive_progress` - with the most fancy animations.
+    * `log` — dumps progress information to a log, console, or text stream; most useful in production, and can optionally include memory and CPU usage stats.
+
+    For more information read the [official documentation](https://dlthub.com/docs/general-usage/pipeline#display-the-loading-progress).
```

Comment:
The anchor link has changed in the docs
```suggestion
    For more information read the [official documentation](https://dlthub.com/docs/general-usage/pipeline#monitor-the-loading-progress).
```

Reply:
also good catch